segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -572,18 +577,15 @@ private long fetchLastAddConfirmed(WriteLedger writeLedger, Map<Long, Long> last
     /**
      * Callback for BookKeeper appends.
      *
-     * @param rc      Response Code.
-     * @param handle  LedgerHandle.
      * @param entryId Assigned EntryId.
-     * @param ctx     Write Context. In our case, the Write we were writing.
+     * @param error   Error.
+     * @param write   the Write we were writing.
      */
-    private void addCallback(int rc, LedgerHandle handle, long entryId, Object ctx) {
-        Write write = (Write) ctx;
+    private void addCallback(Long entryId, Throwable error, Write write) {
         try {
-            assert handle.getId() == write.getWriteLedger().ledger.getId()
-                    : "Handle.Id mismatch: " + write.getWriteLedger().ledger.getId() + " vs " + handle.getId();
-            write.setEntryId(entryId);
-            if (rc == 0) {
+            if (error == null) {
+                assert entryId != null;
+                write.setEntryId(entryId);
```

Comment:
here we are in the success case branch, so we are setting the entryId so that the caller knows it, it is the same behaviour as before.
The `if` was `rc==0` that means success, now the condition is `error==null`

Or maybe I misunderstood your comment

Reply:
This is pre-existing code. The entry id is used by upstream code to calculate an address in the BookKeeper Log (not ledger) which is used for truncation purposes.